Sources/AblyChat/RoomLifecycleManager.swift

```diff
@@ -364,6 +400,23 @@ internal actor RoomLifecycleManager<Contributor: RoomLifecycleContributor> {
 
                 changeStatus(to: .suspended(error: reason))
             }
+        case .attaching:
+            if !hasOperationInProgress, !contributorAnnotations[contributor].hasTransientDisconnectTimeout {
+                // CHA-RL4b7
+                let transientDisconnectTimeout = ContributorAnnotation.TransientDisconnectTimeout()
+                contributorAnnotations[contributor].transientDisconnectTimeout = transientDisconnectTimeout
+                logger.log(message: "Starting transient disconnect timeout \(transientDisconnectTimeout.id) for \(contributor)", level: .debug)
+                transientDisconnectTimeout.task = Task {
+                    do {
+                        try await clock.sleep(timeInterval: 5)
```

Comment:
I think if you later cancel this task you should check if it was cancelled here (after sleeping).

Reply:
Hmm, do you think that's necessary given that the `clock.sleep` call [will throw a `CancellationError` if the task is cancelled](https://developer.apple.com/documentation/swift/task/sleep(nanoseconds:))?

Reply:
If you still need `changeStatus(to:...)` call even if the task was cancelled, then of course no, not needed.

Reply:
I’ve added in a `Task.checkCancellation` just to be on the safe side and make the intention clear 🤷

Reply:
Okay, but now it will be catched as well and `changeStatus` will be called regardless, no different from what it was.

Reply:
OK, I’ve fixed and added a test.